### PR TITLE
feat: add places search api

### PR DESCRIPTION
**Pull request by KnisterPeter**

### CODE REVIEW

File: src/geo-map-places-service-google.ts [hunk 1]
```diff
@@ -1,17 +1,22 @@
 import * as Types from './types';
+import { GeoMap } from './geo-map';
+import { GeoMapPlacesResult } from './geo-map-places-service';
 
 export class GeoMapPlacesServiceGoogle
   implements Types.GeoMapPlacesServiceImplementation {
   private api: Types.GoogleApi;
+  private map: GeoMap;
```
Comment (mabels) [inline src/geo-map-places-service-google.ts]: readonly bitte

File: src/geo-map-places-service-google.ts [hunk 2]
```diff
@@ -25,21 +30,51 @@ export class GeoMapPlacesServiceGoogle
...
+  ): Promise<Types.Result<GeoMapPlacesResult[]>> {
+    const container = document.createElement('div');
+    const service = new this.api.places.PlacesService(container);
+
+    const request: google.maps.places.FindPlaceFromQueryRequest = {
+      query: needle,
+      fields: ['formatted_address', 'name', 'place_id', 'geometry'],
+      locationBias: {
+        center: await this.map.getCenter(),
+        radius: 50000
```
Comment (mabels) [inline src/geo-map-places-service-google.ts]: 50000 why, drop a comment which says search the hole earth!

File: src/geo-map-places-service.ts [hunk 3]
```diff
@@ -9,19 +10,30 @@ export type GeoMapPlacesServiceCreateInit =
 export interface GeoMapPlacesServiceInitGoogle {
   api: Types.GoogleApi;
   type: Types.GeoMapProvider.Google;
+  map: GeoMap;
```
Comment (mabels) [inline src/geo-map-places-service.ts]: readonly?

File: src/geo-map-places-service-google.ts [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
 
   public static create(init: {
     api: Types.GoogleApi;
+    map: GeoMap;
   }): GeoMapPlacesServiceGoogle {
     return new GeoMapPlacesServiceGoogle(init);
   }
 
-  private constructor(init: { api: Types.GoogleApi }) {
+  private constructor(init: { api: Types.GoogleApi; map: GeoMap }) {
```
Comment (mabels) [inline src/geo-map-places-service-google.ts]: remove geomap again

File: src/geo-map-places-service-google.ts [hunk 5]
```diff
@@ -24,4 +29,61 @@ export class GeoMapPlacesServiceGoogle
       });
     });
   }
+
+  public async search(
```
Comment (mabels) [inline src/geo-map-places-service-google.ts]: please add center as a parameter

Comment (mabels) [reply src/geo-map-places-service-google.ts]: center is { center/lat/lan and optional radiusinkm} if radius not set use 50000

**Comment by KnisterPeter** [comment]
@mabels Please review again

File: src/types/geo-map.ts [hunk 6]
```diff
@@ -396,3 +401,14 @@ export interface GeoPlace {
     streetNumber: string | undefined;
   };
 }
+
+export interface GeoMapPlace {
```
Comment (mabels) [inline src/types/geo-map.ts]: atleast here readonly

File: src/geo-map-places-service-google.ts (hunk 5 above)
Comment (mabels) [reply src/geo-map-places-service-google.ts]: the default 50000 is missing and i would suggest to have an interface SearchArea with center und radius as attributes

**Pull request merged**